packages/extension-sdk/src/connect/extension_host_api.ts

```diff
@@ -52,54 +54,72 @@ export class ExtensionHostApiImpl implements ExtensionHostApi {
   private chattyHost: ChattyHostConnection
   private setInitialRoute?: (route: string, routeState?: any) => void
   private hostChangedRoute?: (route: string, routeState?: any) => void
+  private vizualizationDataReceivedCallback?: VizualizationDataReceivedCallback
+
   private contextData?: string
 
   constructor(configuration: ExtensionHostApiConfiguration) {
     this._configuration = configuration
-    const { chattyHost, setInitialRoute, hostChangedRoute } =
-      this._configuration
+    const {
+      chattyHost,
+      setInitialRoute,
+      hostChangedRoute,
+      vizualizationDataReceivedCallback,
+    } = this._configuration
     this.chattyHost = chattyHost
     this.setInitialRoute = setInitialRoute
     this.hostChangedRoute = hostChangedRoute
+    this.vizualizationDataReceivedCallback = vizualizationDataReceivedCallback
   }
 
   get lookerHostData() {
     return this._lookerHostData
   }
 
   handleNotification(
-    message?: ExtensionNotification
+    message: ExtensionNotification
   ): ExtensionInitializationResponse | undefined {
-    const { type, payload } = message || {}
+    const { type } = message
     switch (type) {
-      case ExtensionNotificationType.ROUTE_CHANGED:
+      case ExtensionNotificationType.ROUTE_CHANGED: {
+        const { payload } = message
         if (this.hostChangedRoute && payload) {
-          const { route, routeState } = payload as RouteChangeData
+          const { route, routeState } = payload
           if (route) {
             this.hostChangedRoute(route, routeState)
           }
         }
         return undefined
+      }
+      case ExtensionNotificationType.VISUALIZATION_DATA: {
+        const { payload } = message
+        if (this.vizualizationDataReceivedCallback) {
```

Comment:
here too but I'm sure you'll catch it when you rename refactor

Reply:
Fixed